### PR TITLE
Sync refactor

### PR DESCRIPTION
**Pull request by haramkim-1**
Bring in synchronous DB query changes for writing (querying still in flux)

### CODE REVIEW

File: clientlib/src/DiscreetORM.ts [hunk 1]
```diff
@@ -55,15 +94,47 @@ export class DatabaseORMIO implements DiscreetORMIO {
...
+        let done = false;
+
+        // deasync will hold this function until query returns
+        this.mysql_conn.query(insertString, function (error, results, fields) {
+            if (error) throw error;
+            console.log("inserted ID is: " + results.insertId);
+            id = results.insertId;
+            done = true;
+        });
+        deasync.loopWhile(function() {return !done});
```
Comment (T40TDS) [inline clientlib/src/DiscreetORM.ts:135]: Is this true busy waiting?

File: clientlib/src/DiscreetORM.ts [hunk 2]
```diff
@@ -148,18 +229,24 @@ export function Listener<I extends ObjectListener<any>>(listener: I) {
...
-    let reference_id = toWrite.discreet_orm_id;
-        // TODO: maybe we should change this to an update or insert instead of a delete?
-    let delete_row_template = 'DELETE FROM ?? WHERE ??;';
-    console.log([result_table_name, ("discreet_orm_id = " + reference_id)]);
-    let escaped_command = sqlstring.format(delete_row_template, [result_table_name, ("discreet_orm_id = " + reference_id)]);
-    discreet_sql_io.writeSQL(escaped_command);
-    addRow(toWrite, discreet_sql_io);
+function writeToDB(to_write: any, discreet_sql_io : DiscreetORMIO) {
+    let result_table_name = to_write.constructor.name;
+    let reference_id = to_write.discreet_orm_id;
```
Comment (T40TDS) [inline clientlib/src/DiscreetORM.ts]: Is this an unused variable?

**Review by T40TDS — APPROVED** [review]
Looks very reasonable. Good to merge I believe, after that unused variable is removed.

**Review by TrueshotBarrage — APPROVED** [review]
Looks good

File: clientlib/src/DiscreetORM.ts (hunk 1 above)
Comment (haramkim-1) [reply clientlib/src/DiscreetORM.ts:135]: It supposedly is not - deasync is supposed to just block the thread waiting for an "external" function call

File: clientlib/src/DiscreetORM.ts (hunk 2 above)
Comment (haramkim-1) [reply clientlib/src/DiscreetORM.ts]: looks like that, will fix

**Pull request merged**